sparse_strips/vello_common/src/glyph.rs

```diff
@@ -0,0 +1,183 @@
+        self.run.normalized_coords = coords;
+        self
+    }
+
+    /// Consumes the builder and fills the glyphs with the current configuration.
+    pub fn fill_glyphs(self, glyphs: impl Iterator<Item = &'a Glyph>) {
+        self.renderer
+            .fill_glyphs(Self::prepare_glyphs(&self.run, glyphs));
+    }
+
+    /// Consumes the builder and strokes the glyphs with the current configuration.
+    pub fn stroke_glyphs(self, glyphs: impl Iterator<Item = &'a Glyph>) {
+        self.renderer
+            .stroke_glyphs(Self::prepare_glyphs(&self.run, glyphs));
+    }
+
+    fn prepare_glyphs(
+        run: &GlyphRun,
+        glyphs: impl Iterator<Item = &'a Glyph>,
+    ) -> impl Iterator<Item = PreparedGlyph> {
+        let font = skrifa::FontRef::from_index(run.font.data.as_ref(), run.font.index).unwrap();
+        let outlines = font.outline_glyphs();
+        let size = Size::new(run.font_size);
+        let normalized_coords = run.normalized_coords.as_slice();
+        let hinting_instance = if run.hint {
+            // TODO: Cache hinting instance.
+            if let Some(instance) =
+                HintingInstance::new(&outlines, size, normalized_coords, HINTING_OPTIONS).ok()
+            {
+                Some(instance)
+            } else {
+                None
+            }
+        } else {
+            None
+        };
+        glyphs.filter_map(move |glyph| {
+            let draw_settings = if let Some(hinting_instance) = &hinting_instance {
+                DrawSettings::hinted(&hinting_instance, false)
+            } else {
+                DrawSettings::unhinted(size, normalized_coords)
+            };
+            let outline = outlines.get(GlyphId::new(glyph.id))?;
+            let mut path = OutlinePath(BezPath::new());
+            outline.draw(draw_settings, &mut path).ok()?;
+            let transform = Affine::translate(Vec2::new(glyph.x as f64, glyph.y as f64));
+            Some(PreparedGlyph::Outline(OutlineGlyph {
+                path: path.0,
+                local_transform: transform,
+            }))
+        })
+    }
+}
+
+/// A sequence of glyphs with shared rendering properties.
+#[derive(Clone, Debug)]
+struct GlyphRun {
+    /// Font for all glyphs in the run.
+    pub font: Font,
+    /// Size of the font in pixels per em.
+    pub font_size: f32,
+    /// Normalized variation coordinates for variable fonts.
+    pub normalized_coords: Vec<NormalizedCoord>,
+    /// Controls whether font hinting is enabled.
+    pub hint: bool,
+}
+
+// TODO: Although these are sane defaults, we might want to make them
+// configurable.
+const HINTING_OPTIONS: HintingOptions = HintingOptions {
+    engine: skrifa::outline::Engine::AutoFallback,
+    target: skrifa::outline::Target::Smooth {
+        mode: skrifa::outline::SmoothMode::Lcd,
+        symmetric_rendering: false,
+        preserve_linear_metrics: true,
+    },
+};
+
+struct OutlinePath(BezPath);
+
+impl OutlinePen for OutlinePath {
```

Comment:
@StewartCanva - I noticed you flipped the Y for the curves defined in the font. I needed to do that too. Is that standard?

Reply:
I believe so. The standard coordinate system that font types use is where the y axis points up. Which is the opposite to most 2d graphics systems.

Reply:
Thank you!